format-common/src/main/java/co/cask/hydrator/format/input/TextInputProvider.java

```diff
@@ -28,42 +29,80 @@
  */
 public class TextInputProvider implements FileInputFormatterProvider {
 
+  @Nullable
+  @Override
+  public Schema getSchema(@Nullable String pathField) {
+    return getDefaultSchema(pathField);
+  }
+
   @Override
   public FileInputFormatter create(Map<String, String> properties, @Nullable Schema schema) {
-    if (schema != null) {
-      Schema.Field offsetField = schema.getField("offset");
-      if (offsetField != null) {
-        Schema offsetSchema = offsetField.getSchema();
-        Schema.Type offsetType = offsetSchema.isNullable() ? offsetSchema.getNonNullable().getType() :
-          offsetSchema.getType();
-        if (offsetType != Schema.Type.LONG) {
-          throw new IllegalArgumentException("Type of 'offset' field must be 'long', but found " + offsetType);
-        }
-      }
+    String pathField = properties.get(FileSourceProperties.PATH_FIELD);
+    if (schema == null) {
+      return new TextInputFormatter(getSchema(pathField));
+    }
 
-      Schema.Field bodyField = schema.getField("body");
-      if (bodyField == null) {
-        throw new IllegalArgumentException("Schema for text format must have a field named 'body'");
-      }
-      Schema bodySchema = bodyField.getSchema();
-      Schema.Type bodyType = bodySchema.isNullable() ? bodySchema.getNonNullable().getType() : bodySchema.getType();
-      if (bodyType != Schema.Type.STRING) {
-        throw new IllegalArgumentException("Type of 'body' field must be 'string', but found + " + bodyType);
+    Schema.Field offsetField = schema.getField("offset");
+    if (offsetField != null) {
+      Schema offsetSchema = offsetField.getSchema();
+      Schema.Type offsetType = offsetSchema.isNullable() ? offsetSchema.getNonNullable().getType() :
+        offsetSchema.getType();
+      if (offsetType != Schema.Type.LONG) {
+        throw new IllegalArgumentException(String.format("The 'offset' field must be of type 'long', but found '%s'",
+                                                         offsetType.name().toLowerCase()));
       }
     }
 
-    if (schema == null) {
-      schema = getDefaultSchema(properties.get("pathField"));
+    Schema.Field bodyField = schema.getField("body");
+    if (bodyField == null) {
+      throw new IllegalArgumentException("The schema for the 'text' format must have a field named 'body'");
     }
+    Schema bodySchema = bodyField.getSchema();
```

Comment:
Same here, it is good to add some comments to this method or the javadoc of this class to specify what is the validation logic for the text format.